pkg/allowedcmd/cmd.go

```diff
@@ -0,0 +1,51 @@
+package allowedcmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+)
+
+type AllowedCommand func(ctx context.Context, arg ...string) (*exec.Cmd, error)
+
+func newCmd(ctx context.Context, fullPathToCmd string, arg ...string) *exec.Cmd {
+	return exec.CommandContext(ctx, fullPathToCmd, arg...) //nolint:forbidigo
+}
+
+func validatedCommand(ctx context.Context, knownPath string, arg ...string) (*exec.Cmd, error) {
+	knownPath = filepath.Clean(knownPath)
+
+	if _, err := os.Stat(knownPath); err == nil {
+		return newCmd(ctx, knownPath, arg...), nil
+	}
+
+	// Not found at known location -- return error for darwin and windows.
+	// We expect to know the exact location for allowlisted commands on all
+	// OSes except for a few Linux distros.
+	if !allowSearchPath() {
+		return nil, fmt.Errorf("not found: %s", knownPath)
+	}
+
+	cmdName := filepath.Base(knownPath)
+	if foundPath, err := exec.LookPath(cmdName); err == nil {
+		return newCmd(ctx, foundPath, arg...), nil
+	}
+
+	return nil, fmt.Errorf("%s not found at %s and could not be located elsewhere", cmdName, knownPath)
+}
+
+func allowSearchPath() bool {
+	if runtime.GOOS != "linux" {
+		return false
+	}
+
+	// We only allow searching for binaries in PATH on NixOS
+	if _, err := os.Stat("/etc/NIXOS"); err == nil {
```

Comment:
I'm not sure if we want to autodetect vs making a config file flag. 

Config file is slightly more effort, but also let's people twiddle things. OTOH, I can almost imagine allowing someone to specify an entire dictionary of where to find commands.

Talking about it, I guess what's here seems fine. But there's some chance we'll iterate. It being in this one place makes that pretty smooth

Reply:
This seems like it should be reliable (per [this](https://discourse.nixos.org/t/detect-that-bash-script-is-running-under-nixos/11402/2)).

Maybe leave this as-is for now, and we can iterate once we're further into NixOS support and know better whether this vs a config file flag makes more sense?

Reply:
Yah. I think leaving for now, and then we can iterate when we know more.